### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by PooyaHekmati**
### Context

Issue(s) closed by this pull request: closes N/A

- [x] The [changelog](https://docs.google.com/spreadsheets/d/1U-3igxdstHTFb6k5dVcE5-x9IL0r6q18eUiwwytKZyA/edit#gid=0) has been updated.
      
How big are the changes in the PR? Would you nominate it for a major version upgrade?
- [ ] Yes
- [x] No

### What

Removes unused test cases.
Removes `Output_Manger.dump_variables`.

### Why

1. To make the metrics and results of running the test suite more accurate.
2. To speed up the test suite.
3. To make the code base easier to navigate and maintain.

### How

Removes the unused functions.
Removes `deprecated` from requirements as it is no longer needed.

### Test plan

Test Suite

### CODE REVIEW

File: tests/test_misc.py
Comment (Isgotkowitz) [inline tests/test_misc.py:1]: Could this class be renamed to `test_general_constants.py`? Seems like it would be a more accurate name

**Review by Isgotkowitz — COMMENTED** [review]
LG8TM. One suggestion about renaming one of the test files

**Review by Isgotkowitz — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [![Flake8](https://img.shields.io/badge/Flake8-passed-brightgreen)](https://github.com/RuminantFarmSystems/MASM/actions/workflows/combined_format_lint_test_mypy.yml)
 [![Pytest](https://img.shields.io/badge/Pytest-passed-brightgreen)](https://github.com/RuminantFarmSystems/MASM/actions/workflows/combined_format_lint_test_mypy.yml)
-[![Coverage](https://img.shields.io/badge/Coverage-95%25-brightgreen)](https://github.com/RuminantFarmSystems/MASM/actions/workflows/combined_format_lint_test_mypy.yml)
-[![Mypy](https://img.shields.io/badge/Mypy-4024%20errors-red)](https://github.com/RuminantFarmSystems/MASM/actions/workflows/combined_format_lint_test_mypy.yml)
+[![Coverage](https://img.shields.io/badge/Coverage-94%25-green)](https://github.com/RuminantFarmSystems/MASM/actions/workflows/combined_format_lint_test_mypy.yml)
```
Comment (allisterakun) [inline README.md]: Why would the test coverage decrease by 1%?

**Review by allisterakun — APPROVED** [review]
LGTM, just one comment about the decrease in test coverage.

File: README.md (hunk 1 above)
Comment (PooyaHekmati) [reply README.md]: It was due to syncing issues, it's gone now.

File: tests/test_misc.py
Comment (PooyaHekmati) [reply tests/test_misc.py:1]: good catch

**Pull request merged**